[BUG] index is -334 in getPodLock

<!--
Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
-->

**What happened**:
```
I0813 06:26:27.700038       7 node.go:138] NodePublishVolume: volume context: map[capacity:1073741824 csi.storage.k8s.io/ephemeral:false csi.storage.k8s.io/pod.name:app-dynamic-cache-umount csi.storage.k8s.io/pod.namespace:default csi.storage.k8s.io/pod.uid:e421992d-49f2-40e1-8bff-7243ad2efcba csi.storage.k8s.io/serviceAccount.name:default juicefs/clean-cache:true juicefs/mount-cpu-limit:5 juicefs/mount-cpu-request:100m juicefs/mount-memory-limit:5Gi juicefs/mount-memory-request:500Mi storage.kubernetes.io/csiProvisionerIdentity:1723529879902-8081-csi.juicefs.com subPath:pvc-3f3b2c72-01f3-4ceb-9b50-9a4c03d14921]
I0813 06:26:27.700070       7 node.go:148] NodePublishVolume: mounting juicefs with secret [bucket metaurl name secret-key storage access-key], options [buffer-size=300 cache-size=100 enable-xattr cache-dir=/mnt/dynamic/cache1:/mnt/dynamic/cache2 free-space-ratio=0.01]
I0813 06:26:27.700104       7 client.go:327] Get pv pvc-3f3b2c72-01f3-4ceb-9b50-9a4c03d14921
I0813 06:26:27.706481       7 client.go:383] Get pvc pod-ce-pvc-dynamic-cache-umount in namespace default
I0813 06:26:27.711275       7 config.go:369] volume  using patch: &{PVCSelector:<nil> CEMountImage: EEMountImage: Image: Labels:map[] Annotations:map[] HostNetwork:<nil> HostPID:<nil> LivenessProbe:nil ReadinessProbe:nil StartupProbe:nil Lifecycle:&Lifecycle{PostStart:nil,PreStop:&Handler{Exec:&ExecAction{Command:[sh -c +e umount  -l; rmdir ; exit 0],},HTTPGet:nil,TCPSocket:nil,},} Resources:nil TerminationGracePeriodSeconds:<nil> Volumes:[] VolumeDevices:[] VolumeMounts:[] Env:[]}
I0813 06:26:27.711403       7 juicefs.go:991] ceFormat cmd: [/usr/local/bin/juicefs format --storage=s3 --bucket=http://juicefs-bucket.minio.default.svc.cluster.local:9000 --access-key=minioadmin --secret-key=${secretkey} ${metaurl} ce-secret]
I0813 06:26:27.711434       7 juicefs.go:422] Get uniqueId of volume [pvc-3f3b2c72-01f3-4ceb-9b50-9a4c03d14921]: pvc-3f3b2c72-01f3-4ceb-9b50-9a4c03d14921
I0813 06:26:27.808229       7 juicefs.go:438] Get uuid of volume [pvc-3f3b2c72-01f3-4ceb-9b50-9a4c03d14921]: ff312dac-d3df-46da-bd46-78ef9a66ce30
I0813 06:26:27.808547       7 pod_mount.go:667] jfsSetting hash: 470153671e028c49ca8f74ab2a6bd4e5669e6f29fd5322f4ced0d4121098126
panic: runtime error: index out of range [-334]

goroutine 213 [running]:
github.com/juicedata/juicefs-csi-driver/pkg/config.GetPodLock({0xcae6140, 0x3f})
	/root/juicefs-csi-driver/pkg/config/config.go:165 +0x83
github.com/juicedata/juicefs-csi-driver/pkg/juicefs/mount.(*PodMount).JMount.func1({0xcae6140, 0x3f}, 0xcc41870, 0xcc41878, 0xcc5cd68, {0x98954dc, 0xcc62960}, 0xcbb70e8, 0xcb4f6e0)
	/root/juicefs-csi-driver/pkg/juicefs/mount/pod_mount.go:70 +0x4b
github.com/juicedata/juicefs-csi-driver/pkg/juicefs/mount.(*PodMount).JMount(0xcc5cd68, {0x98954dc, 0xcc62960}, 0xcb4f6e0, 0xcbb70e8)
	/root/juicefs-csi-driver/pkg/juicefs/mount/pod_mount.go:92 +0x1a7
github.com/juicedata/juicefs-csi-driver/pkg/juicefs.(*juicefs).MountFs(0xc988280, {0x98954dc, 0xcc62960}, 0xcb4f6e0, 0xcbb70e8)
	/root/juicefs-csi-driver/pkg/juicefs/juicefs.go:896 +0x192
github.com/juicedata/juicefs-csi-driver/pkg/juicefs.(*juicefs).JfsMount(0xc988280, {0x98954dc, 0xcc62960}, {0xc8c6090, 0x28}, {0xc8c8140, 0x9c}, 0xcd4afe0, 0xcd4b000, {0xcac1020, ...})
	/root/juicefs-csi-driver/pkg/juicefs/juicefs.go:334 +0xfd
github.com/juicedata/juicefs-csi-driver/pkg/driver.(*nodeService).NodePublishVolume(0xcc6c400, {0x98954dc, 0xcc62960}, 0xc9b2d80)
	/root/juicefs-csi-driver/pkg/driver/node.go:149 +0x88f
github.com/container-storage-interface/spec/lib/go/csi._Node_NodePublishVolume_Handler.func1({0x98954dc, 0xcc62960}, {0x95a5ce0, 0xc9b2d80})
	/root/go/pkg/mod/github.com/container-storage-interface/spec@v1.5.0/lib/go/csi/csi.pb.go:6146 +0x73
github.com/juicedata/juicefs-csi-driver/pkg/driver.(*Driver).Run.func1({0x98954dc, 0xcc62960}, {0x95a5ce0, 0xc9b2d80}, 0xcc66b70, 0xcc66b80)
	/root/juicefs-csi-driver/pkg/driver/driver.go:100 +0x40
github.com/container-storage-interface/spec/lib/go/csi._Node_NodePublishVolume_Handler({0x96300e0, 0xcc6c3f0}, {0x98954dc, 0xcc62960}, 0xc9b2d40, 0x970df04)
	/root/go/pkg/mod/github.com/container-storage-interface/spec@v1.5.0/lib/go/csi/csi.pb.go:6148 +0x17e
google.golang.org/grpc.(*Server).processUnaryRPC(0xcbb6c68, {0x98991b4, 0xca607e8}, 0xcc16848, 0xcc5d260, 0xa4ed3b8, 0x0)
	/root/go/pkg/mod/google.golang.org/grpc@v1.56.3/server.go:1335 +0xea4
google.golang.org/grpc.(*Server).handleStream(0xcbb6c68, {0x98991b4, 0xca607e8}, 0xcc16848, 0x0)
	/root/go/pkg/mod/google.golang.org/grpc@v1.56.3/server.go:1712 +0xa13
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	/root/go/pkg/mod/google.golang.org/grpc@v1.56.3/server.go:947 +0xe3
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 208
	/root/go/pkg/mod/google.golang.org/grpc@v1.56.3/server.go:958 +0x13e
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**

**Environment**:
- JuiceFS CSI Driver version (which image tag did your CSI Driver use):
- Kubernetes version (e.g. `kubectl version`):
- Object storage (cloud provider and region):
- Metadata engine info (version, cloud provider managed or self maintained):
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:

--- a/pkg/config/config.go
+++ b/pkg/config/config.go
@@ -161,8 +161,8 @@ var PodLocks [1024]sync.Mutex
 func GetPodLock(podHashVal string) *sync.Mutex {
 	h := fnv.New32a()
 	h.Write([]byte(podHashVal))
-	index := int(h.Sum32())
-	return &PodLocks[index%1024]
+	index := h.Sum32() % 1024
+	return &PodLocks[index]
 }
 
 func MustGetWebPort() int {

